examples/demo/src/orders/OrderList.tsx

```diff
@@ -0,0 +1,239 @@
+                <MobileGrid {...rest} ids={selectedIds} />
+            ) : (
+                <div>
+                    {filterValues.status === 'ordered' && (
+                        <Datagrid
+                            {...rest}
+                            ids={ordered}
+                            optimized
+                            rowClick="edit"
+                        >
+                            <DateField source="date" showTime />
+                            <TextField source="reference" />
+                            <CustomerReferenceField />
+                            <NbItemsField />
+                            <NumberField
+                                source="total"
+                                options={{
+                                    style: 'currency',
+                                    currency: 'USD',
+                                }}
+                                className={classes.total}
+                            />
+                        </Datagrid>
+                    )}
+                    {filterValues.status === 'delivered' && (
+                        <Datagrid {...rest} ids={delivered}>
+                            <DateField source="date" showTime />
+                            <TextField source="reference" />
+                            <CustomerReferenceField />
+                            <NbItemsField />
+                            <NumberField
+                                source="total"
+                                options={{
+                                    style: 'currency',
+                                    currency: 'USD',
+                                }}
+                                className={classes.total}
+                            />
+                            <BooleanField source="returned" />
+                            <EditButton />
+                        </Datagrid>
+                    )}
+                    {filterValues.status === 'cancelled' && (
+                        <Datagrid {...rest} ids={cancelled}>
+                            <DateField source="date" showTime />
+                            <TextField source="reference" />
+                            <CustomerReferenceField />
+                            <NbItemsField />
+                            <NumberField
+                                source="total"
+                                options={{
+                                    style: 'currency',
+                                    currency: 'USD',
+                                }}
+                                className={classes.total}
+                            />
+                            <BooleanField source="returned" />
+                            <EditButton />
+                        </Datagrid>
+                    )}
+                </div>
+            )}
+        </Fragment>
+    );
+};
+
+const StyledTabbedDatagrid: FC<DatagridProps<Order>> = props => {
```

Comment:
I don't think you need a separate component anymore. It was done so that we could use hooks while the `TabbedDatagrid` component was class based. Now that it's a function, you can move the below hooks to the main component.

Reply:
Fixed